src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -320,38 +325,97 @@ const NetworkAssetSelectInput = ({
     };
     handleBalanceGet();
-  }, [supportedChains, sdk, walletAddress]);
+  }, [supportedChains, sdk, accountAddress]);
+
+  useEffect(() => {
+    const handleKeybasedBalanceGet = async () => {
+      if (!sdk || !providerAddress) return;
+      await getKeybasedWalletBalancesPerChain(providerAddress, supportedChains);
+    };
+    handleKeybasedBalanceGet();
+  }, [supportedChains, sdk, providerAddress]);
 
   const filteredSelectedNetworkAssets = useMemo(() => {
     const filtered = selectedNetworkAssets.filter((asset) => containsText(asset.name, assetSearchQuery) || containsText(asset.symbol, assetSearchQuery));
     return orderBy(filtered, [(asset) => asset.balanceWorthUsd ?? 0, 'name'], ['desc', 'asc']);
   }, [selectedNetworkAssets, assetSearchQuery]);
 
   useEffect(() => {
-    const updateAvalancheBalance = () => {
+    const updateAvalancheBalanceSmart = () => {
       if (!sdk ||
         !walletAddress ||
         !preselectedNetwork ||
         preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
         !smartWalletBalanceByChain ||
-        filteredSelectedNetworkAssets.length
+        !filteredSelectedNetworkAssets.length ||
+        isLoadingAssets
       ) {
         return;
       }
       setsmartWalletBalanceByChain([
-        ...smartWalletBalanceByChain,
+        ...smartWalletBalanceByChain.filter(
+          (element) => element.chain !== CHAIN_ID.AVALANCHE
+        ),
         {
           total: sumAssetsBalanceWorth(filteredSelectedNetworkAssets),
-          title: (supportedChains.find(({ chainId }) =>  chainId === CHAIN_ID.AVALANCHE) as Chain).title,
+          title: (
+            supportedChains.find(
+              ({ chainId }) => chainId === CHAIN_ID.AVALANCHE
+            ) as Chain
+          ).title,
           chain: CHAIN_ID.AVALANCHE,
         },
       ]);
     };
-    updateAvalancheBalance();
+    updateAvalancheBalanceSmart();
   }, [
     sdk,
     supportedChains,
-    walletAddress,
+    accountAddress,
+    preselectedNetwork,
+    filteredSelectedNetworkAssets,
+    isLoadingAssets,
+  ]);
+
+  useEffect(() => {
+    console.log("smartWalletBalanceByChain",smartWalletBalanceByChain)
+  }, [smartWalletBalanceByChain])
+  
+
+  useEffect(() => {
+    const updateAvalancheBalanceKeybased = () => {
+      if (
+        !sdk ||
+        !walletAddress ||
+        !preselectedNetwork ||
+        preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
+        !keybasedWalletBalanceByChain ||
+        !filteredSelectedNetworkAssets.length ||
+        accountType !== DestinationWalletEnum.Key ||
+        isLoadingAssets
+      )
+        return;
+
+      setKeybasedWalletBalanceByChain([
+        ...keybasedWalletBalanceByChain.filter(
+          (element) => element.chain !== CHAIN_ID.AVALANCHE
+        ),
+        {
+          total: filteredSelectedNetworkAssets.reduce((acc, curr) => {
+            return curr.balanceWorthUsd ? acc + curr.balanceWorthUsd : acc;
+          }, 0),
```

Comment:
Could probably shorten this, and I added `?` just to be safe. Also could you change `curr` to a more descriptive name, like `asset` or something

```suggestion
          total: filteredSelectedNetworkAssets.reduce((acc, curr) => curr?.balanceWorthUsd ? acc + curr.balanceWorthUsd : acc, 0),
```